### PR TITLE
add test for "Better Method#inspect" introduced in Ruby 2.7

### PR DESCRIPTION
**Pull request by lxxxvi**
Hello 👋 

From #745 

> * [ ] Method#inspect shows more information. Feature #14145

I hope I've chosen the right approach and put it in the right place?!

### CODE REVIEW

**Comment by andrykonchin** [comment]
Looks good 👍 .

But I would add some corner cases with `*a`, `**b`, `&c` etc. They could be based on specs from Ruby own tests https://github.com/ruby/ruby/pull/2618/files#diff-4ed63f6118a737e9326cb7c64b07a527R635-R657.

**Comment by eregon** [comment]
There are quite few methods in https://github.com/ruby/spec/blob/bfd843a4d8e934ebb5994eaba3c6176a91778082/core/method/fixtures/classes.rb#L49-L79
It would be good to test the output for each "kind" of argument (pre/opt/rest/post/keyreq/keyopt/keyrest/block).

File: core/method/shared/to_s.rb [hunk 1]
```diff
@@ -24,6 +24,13 @@
     @string.should =~ /\#bar/
   end
 
+  ruby_version_is "2.7" do
+    it "returns a String containing method arguments" do
+      m = MethodSpecs::Methods.new.method :two_req
+      m.send(@method).should =~ /\#two_req\(a, b\)/
```
Comment (eregon) [inline core/method/shared/to_s.rb:30]: You can use `m.send(@method).should.include?("two_req(a, b)")`.
That way, there is no need to escape any character.

**Comment by lxxxvi** [comment]
Hello @andrykonchin and @eregon

Thank you for your feedbacks. :tada:

Extending the tests revealed a new trivial issue: line size.
First I came up with this, but I think it's not very readable.

```ruby
it "returns a String containing method arguments" do
  obj = MethodSpecs::Methods.new
  obj.method(:zero).send(@method).should.include?("#zero()")
  obj.method(:one_req).send(@method).should.include?("one_req(a)")
  obj.method(:two_req).send(@method).should.include?("two_req(a, b)")
  # ... there are more ...
  obj.method(:one_req_one_opt_with_splat_and_block).send(@method).should.include?("one_req_one_opt_with_splat_and_block(a, b=..., *c, &blk)")
  obj.method(:two_req_one_opt_with_splat_and_block).send(@method).should.include?("two_req_one_opt_with_splat_and_block(a, b, c=..., *d, &blk)")
  obj.method(:one_req_two_opt_with_splat_and_block).send(@method).should.include?("one_req_two_opt_with_splat_and_block(a, b=..., c=..., *d, &blk)")
end
```

Putting the inputs into a "table" and then loop makes it somewhat more readable, but slower (I guess?).

```ruby
it "returns a String containing method arguments" do
  obj = MethodSpecs::Methods.new
  [
    [:zero, "#zero()"],
    [:one_req, "one_req(a)"],
    [:two_req, "two_req(a, b)"],
    # ... there are more ...
    [:one_req_one_opt_with_splat_and_block, "one_req_one_opt_with_splat_and_block(a, b=..., *c, &blk)"],
    [:two_req_one_opt_with_splat_and_block, "two_req_one_opt_with_splat_and_block(a, b, c=..., *d, &blk)"],
    [:one_req_two_opt_with_splat_and_block, "one_req_two_opt_with_splat_and_block(a, b=..., c=..., *d, &blk)"]
  ].each do |method_key, expected_string|
    obj.method(method_key).send(@method).should.include?(expected_string)
  end
end
```

Do you have any preferences here?
Or should I put all assertions into indiviual tests?

**Comment by eregon** [comment]
I think we can just omit the method name in the String given to `include?`:
```ruby
it "returns a String containing method arguments" do
  obj = MethodSpecs::Methods.new
  obj.method(:zero).send(@method).should.include?("()")
  obj.method(:one_req).send(@method).should.include?("(a)")
  obj.method(:two_req).send(@method).should.include?("(a, b)")
  # ... there are more ...
  obj.method(:one_req_one_opt_with_splat_and_block).send(@method).should.include?("(a, b=..., *c, &blk)")
  obj.method(:two_req_one_opt_with_splat_and_block).send(@method).should.include?("(a, b, c=..., *d, &blk)")
  obj.method(:one_req_two_opt_with_splat_and_block).send(@method).should.include?("(a, b=..., c=..., *d, &blk)")
end
```

The table gives less information in the case an expectation fails, so it's not as good.

Another possibility would be to just break lines like:
```ruby
it "returns a String containing method arguments" do
  obj.method(:one_req_one_opt_with_splat_and_block).send(@method)
    .should.include?("one_req_one_opt_with_splat_and_block(a, b=..., *c, &blk)")
end
```

As mentioned in https://github.com/ruby/spec/pull/793#issuecomment-705084526 I think it's useful to test each kind of argument, but not all combinations, so the 3 `*_req_*_opt_with_splat_and_block` are redundant and only one of them is useful to test.

**Comment by lxxxvi** [comment]
Hello @eregon

Once again, thank you for your review and feedback!

> I think we can just omit the method name in the String given to `include?`

Perfect, done. It fits. :rocket:

However, I'm bit intimidated by the comment for testing _"each kind of argument"_, as I am insecure what some of them mean 🙃 

> _"pre/opt/rest/post/keyreq/keyopt/keyrest/block"_

However, I added tests for each simple "kind" of argument, plus one "complicated" case.
Please let me know if you're missing some cases.

Thanks

**Review by eregon — APPROVED** [review]
This looks great, thank you!
